abci/server/errors.go

```diff
@@ -0,0 +1,33 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/cometbft/cometbft/abci/types"
+)
+
+// ErrUnknownServerType is returned when trying to create a server with invalid transport option.
+type ErrUnknownServerType struct {
+	ServerType string
+}
+
+func (e ErrUnknownServerType) Error() string {
+	return fmt.Sprintf("Unknown server type %s", e.ServerType)
+}
+
+// ErrConnectionNotExists is returned when trying to access non-existent network connection
+type ErrConnectionNotExists struct {
+	ConnID int
+}
+
+func (e ErrConnectionNotExists) Error() string {
+	return fmt.Sprintf("Connection %d does not exist", e.ConnID)
+}
+
+type ErrUnknownClientRequest struct {
```

Comment:
`ErrUnexpectedRequest` perhaps?

Reply:
wdyt of `ErrUnknownRequest`? I am changing this: `"unknown request from client %T"` with this error

Reply:
@thanethomson